src/Umbraco.Infrastructure/Security/MemberRoleStore.cs

```diff
@@ -0,0 +1,342 @@
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            return Task.FromResult(role.Id);
+        }
+
+        /// <inheritdoc />
+        public Task<string> GetRoleNameAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            return Task.FromResult(role.Name);
+        }
+
+        /// <inheritdoc />
+        public Task SetRoleNameAsync(TRole role, string roleName, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            if (!int.TryParse(role.Id, out int roleId))
+            {
+                //TODO: what identity error should we return in this case?
+                return Task.FromResult(IdentityResult.Failed(ErrorDescriber.DefaultError()));
+            }
+
+            IMemberGroup memberGroup = _memberGroupService.GetById(roleId);
+
+            if (memberGroup != null)
+            {
+                //TODO: confirm logic
+                memberGroup.Name = roleName;
+                _memberGroupService.Save(memberGroup);
+                role.Name = roleName;
+            }
+            else
+            {
+                return Task.FromResult(IdentityResult.Failed(_memberGroupNotFoundError));
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public Task<string> GetNormalizedRoleNameAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            //TODO: are we utilising NormalizedRoleName?
+            return Task.FromResult(role.NormalizedName);
+        }
+
+        /// <inheritdoc />
+        public Task SetNormalizedRoleNameAsync(TRole role, string normalizedName, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            //TODO: are we utilising NormalizedRoleName and do we need to set it in the memberGroupService?
+            role.NormalizedName = normalizedName;
```

Comment:
**suggestion:** I would treat name and normalized name as one and the same for now. I think that's what users is doing

Reply:
Thanks - I have changed to just call the Get/ Set Name methods

Reply:
Leaving conversation open to confirm with Umbraco HQ